debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerConnectorTask.java

```diff
@@ -97,7 +97,8 @@ public ChangeEventSourceCoordinator<SqlServerPartition, SqlServerOffsetContext>
                 .loggingContextSupplier(() -> taskContext.configureLoggingContext(CONTEXT_NAME))
                 .build();
 
-        errorHandler = new SqlServerErrorHandler(connectorConfig, queue);
+        int retries = (errorHandler == null) ? 0 : errorHandler.getRetries();
+        errorHandler = new SqlServerErrorHandler(connectorConfig, queue, retries, connectorConfig.getMaxRetries());
```

Comment:
Why si this necessary? Can't you just reuse the exising handler? Then you will not need to pass `retries`.